services/errors.go

```diff
@@ -132,3 +147,16 @@ var (
 		ErrUnableToGetNodeStatus,
 	}
 )
+
+func GetDetailedError(proto *types.Error, cause error) *types.Error {
```

Comment:
```suggestion
func NewDetailedError(proto *types.Error, cause error) *types.Error {
```
Also add a documentation comment.

Reply:
ok